src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -90,6 +91,41 @@ def get_helm_registry(profile, location):
     cred, _, _ = profile.get_login_credentials(
         resource='https://management.core.windows.net/')
     token = cred._token_retriever()[2].get('accessToken')  # pylint: disable=protected-access
+    get_chart_location_url = "https://{}.dp.kubernetesconfiguration.azure.com/{}/GetLatestHelmPackagePath?api-version=2019-11-01-preview".format(location, 'azure-arc-k8sagents')
+    query_parameters = {}
+    query_parameters['releaseTrain'] = os.getenv('RELEASETRAIN') if os.getenv('RELEASETRAIN') else 'stable'
+    header_parameters = {}
+    header_parameters['Authorization'] = "Bearer {}".format(str(token))
+    try:
+        response = requests.post(get_chart_location_url, params=query_parameters, headers=header_parameters)
+    except Exception as e:
+        telemetry.set_exception(exception=e, fault_type=consts.Get_HelmRegistery_Path_Fault_Type,
+                                summary='Error while fetching helm chart registry path')
+        raise CLIError("Error while fetching helm chart registry path: " + str(e))
+    if response.status_code == 200:
+        return response.json().get('repositoryPath')
+    telemetry.set_exception(exception=str(response.json()), fault_type=consts.Get_HelmRegistery_Path_Fault_Type,
+                            summary='Error while fetching helm chart registry path')
+    raise CLIError("Error while fetching helm chart registry path: {}".format(str(response.json())))
+
+
+def get_helm_registry_using_token(values_file, location):
```

Comment:
Can we have an implementation where we read the values file only once? Basically merge the functions 'get_flag_from_values_file' and 'get_helm_registry_using_token'

Reply:
Okay, will look into it

Reply:
Done